drivers/mcp2515/candev_mcp2515.c

```diff
@@ -451,13 +454,27 @@ static int _set_filter(candev_t *dev, const struct can_filter *filter)
         f.can_mask &= CAN_SFF_MASK;
     }
 
+    uint8_t rx_mailbox = 0;
```

Comment:
Am I interpreting it correctly that you removed an exposed member `mailbox` and replaced it with a hidden one? What's the impact of this?

Reply:
The idea is to have a generic can_filter struct for all CAN controllers. By adding the `can_filter_conf_t`struct (which will define where the filter will be applied, disable the filter or reject CAN messages with the specified filter i.e same as mailbox but with other features extra), the former specific member `mailbox` of the MCP2515 will be defined according to the configuration provided.